### PR TITLE
ruby: Implement getting appid

### PR DESCRIPTION
**Pull request by mpozniak95**
## Related Issue
https://github.com/Granulate/gprofiler/issues/505

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -487,7 +488,8 @@ def assert_app_id(application_pid: int, runtime: str, in_container: bool) -> Gen
     desired_name_and_getter = {
         "java": (get_java_app_id, "java: Fibonacci.jar"),
         "python": (get_python_app_id, "python: lister.py (/app/lister.py)"),
-        "node": (get_node_app_id, "nodejs: fibonacci.js (/app/fibonacci.js"),
```
Comment (Jongy) [inline tests/conftest.py:490]: How did we merge it this way o.O I mean how did the tests pass

File: gprofiler/metadata/application_identifiers.py [hunk 2]
```diff
@@ -290,6 +290,21 @@ def get_app_id(self, process: Process) -> Optional[str]:
...
 
 
+class _RubyModuleApplicationIdentifier(_ApplicationIdentifier):
+    def get_app_id(self, process: Process) -> Optional[str]:
+        skip_next = False
+        for arg in process.cmdline():
+            if skip_next:
+                skip_next = False
+                continue
+            if arg == "-r":
```
Comment (Jongy) [inline gprofiler/metadata/application_identifiers.py:300]: I read that these 2 are equal:
```
ruby -r require.rb app.rb
ruby -rrequire.rb app.rb
```

so you need to check if anything follows the `-r` and if so, don't skip next and only skip the current argument.

**Review by Jongy — CHANGES_REQUESTED** [review]
One small comment :)

File: tests/conftest.py (hunk 1 above)
Comment (mpozniak95) [reply tests/conftest.py:490]: So this was never tested :(

Comment (Jongy) [reply tests/conftest.py:490]: Ohh.

File: gprofiler/metadata/application_identifiers.py [hunk 3]
```diff
@@ -290,6 +290,23 @@ def get_app_id(self, process: Process) -> Optional[str]:
...
 
 
+class _RubyModuleApplicationIdentifier(_ApplicationIdentifier):
+    def get_app_id(self, process: Process) -> Optional[str]:
+        skip_next = False
+        for arg in process.cmdline():
+            if skip_next:
+                skip_next = False
+                continue
+            if re.compile(r"-r.+").match(arg):
```
Comment (Jongy) [inline gprofiler/metadata/application_identifiers.py:300]: Simpler - if arg startswith "-r". Covers both cases so we won't need the ==.

File: gprofiler/metadata/application_identifiers.py [hunk 4]
```diff
@@ -290,6 +290,22 @@ def get_app_id(self, process: Process) -> Optional[str]:
...
 
+class _RubyModuleApplicationIdentifier(_ApplicationIdentifier):
+    def get_app_id(self, process: Process) -> Optional[str]:
+        skip_next = False
+        for arg in process.cmdline():
+            if skip_next:
+                skip_next = False
+                continue
+            if arg.startswith("-r"):
+                if len(arg) <= 2:
```
Comment (Jongy) [inline gprofiler/metadata/application_identifiers.py:301]: Could be `== 2` because it's always at least 2 due to the startswith, but meh

**Review by Jongy — APPROVED** [review]

**Pull request merged**